eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxFunder.scala

```diff
@@ -0,0 +1,479 @@
+          sign(txWithWitnessData, htlcFeerate, htlcTx.txInfo.amountIn)
+        } else {
+          addWalletInputs(htlcTx, targetFeerate)
+        }
+      case claimHtlcTx: ClaimHtlcWithWitnessData =>
+        adjustClaimHtlcTxOutput(claimHtlcTx, targetFeerate, cmd.commitments.localParams.dustLimit) match {
+          case Left(reason) =>
+            // The htlc isn't economical to claim at the current feerate, but if the feerate goes down, we may want to claim it later.
+            log.warn("skipping {}: {} (feerate={})", cmd.desc, reason, targetFeerate)
+            replyTo ! FundingFailed(TxPublisher.TxRejectedReason.TxSkipped(retryNextBlock = true))
+            Behaviors.stopped
+          case Right(updatedClaimHtlcTx) =>
+            sign(updatedClaimHtlcTx, targetFeerate, updatedClaimHtlcTx.txInfo.amountIn)
+        }
+    }
+  }
+
+  def bump(previousTx: FundedTx, targetFeerate: FeeratePerKw): Behavior[Command] = {
+    adjustPreviousTxOutput(previousTx, targetFeerate, cmd.commitments) match {
+      case AdjustPreviousTxOutputResult.Skip(reason) =>
+        log.warn("skipping {} fee bumping: {} (feerate={})", cmd.desc, reason, targetFeerate)
+        replyTo ! FundingFailed(TxPublisher.TxRejectedReason.TxSkipped(retryNextBlock = true))
+        Behaviors.stopped
+      case AdjustPreviousTxOutputResult.TxOutputAdjusted(updatedTx) =>
+        log.debug("bumping {} fees without adding new inputs: txid={}", cmd.desc, updatedTx.txInfo.tx.txid)
+        sign(updatedTx, targetFeerate, previousTx.totalAmountIn)
+      case AdjustPreviousTxOutputResult.AddWalletInputs(tx) =>
+        log.debug("bumping {} fees requires adding new inputs (feerate={})", cmd.desc, targetFeerate)
+        // We restore the original transaction (remove previous attempt's wallet inputs).
+        val resetTx = tx.updateTx(cmd.txInfo.tx)
+        addWalletInputs(resetTx, targetFeerate)
+    }
+  }
+
+  def addWalletInputs(txWithWitnessData: ReplaceableTxWithWalletInputs, targetFeerate: FeeratePerKw): Behavior[Command] = {
+    context.pipeToSelf(addInputs(txWithWitnessData, targetFeerate, cmd.commitments)) {
+      case Success((fundedTx, totalAmountIn)) => AddInputsOk(fundedTx, totalAmountIn)
+      case Failure(reason) => AddInputsFailed(reason)
+    }
+    Behaviors.receiveMessagePartial {
+      case AddInputsOk(fundedTx, totalAmountIn) =>
+        log.info("added {} wallet input(s) and {} wallet output(s) to {}", fundedTx.txInfo.tx.txIn.length - 1, fundedTx.txInfo.tx.txOut.length - 1, cmd.desc)
+        sign(fundedTx, targetFeerate, totalAmountIn)
+      case AddInputsFailed(reason) =>
+        if (reason.getMessage.contains("Insufficient funds")) {
+          val nodeOperatorMessage =
+            s"""Insufficient funds in bitcoin wallet to set feerate=$targetFeerate for ${cmd.desc}.
+               |You should add more utxos to your bitcoin wallet to guarantee funds safety.
```

Comment:
```suggestion
               |You should add more utxos to your bitcoin wallet to guarantee funds safety.
               |Attempts will be made periodically to re-publish this transaction.
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/2113/commits/67e0d962f11ca024f2b1b84e9eecd7b74753b4b6